src/adminUser.service.ts

```diff
@@ -12,8 +12,14 @@ import AdminUser from './adminUser.entity'
 import { DuplicateEmailException } from './exceptions/userAdmin.exception'
 import { AdminUserValidationException } from './exceptions/adminUserValidation.exception'
 
+type AdminUserWithoutPassword = Omit<AdminUser, 'password'>
+
+export interface AdminUserServiceInterface extends EntitySubscriberInterface<AdminUser> {
+  validateCredentials(email: string, pass: string): Promise<AdminUserWithoutPassword | null>
+}
```

Comment:
The whole point of this interface is that it's supposed to be entity-agnostic, so it's not aware of AdminUsers (it could be implemented for any sort of User entity). You'll need to:

- Rename it something like `AdminAuthenticatorInterface`
- Move it to a file (adminAuthenticator.interface)
- Export it from the library (root index.ts)
- Change the signature: `validateAdminCredentials(username: string, password: string) => object | null | Promise<object | null>`